### ISSUE TITLE
Scaling portal causes game to completle freeze.

### ISSUE THREAD

**Issue by Asaak232** [qouteall/ImmersivePortalsMod#1159]
### According to your testing, does the issue occur with ONLY Immersive Portals mod and Farbic API?

No

### Describe the issue

Scaling portal causes game to completle freeze. There is no crashlog, i just go through the portal, game completly dies, and after i manually kill the process, and enter the same world, only to find out that my size changed, and i am on the other side of a portal. All pehkui console commands working just fine. It only occurse when i use portal. I use 1.19.2 latest version of immersive portals, so as pehkui, gravity api and miniscaled. There is also world edit, but i tried without it, changed nothing. 

### Post the log here

```shell
[03:48:02] [main/INFO]: Loading Minecraft 1.19.2 with Fabric Loader 0.14.9
[03:48:02] [main/INFO]: Loading 63 mods:
	- cardinal-components-base 4.1.4 via gravity_api
	- cardinal-components-entity 4.1.4 via gravity_api
	- cloth-basic-math 0.6.1 via cloth-config
	- cloth-config 7.0.72 via immersive_portals
	- fabric-api 0.59.0+1.19.2
	- fabric-api-base 0.4.10+e62f51a390 via fabric-api
	- fabric-api-lookup-api-v1 1.6.8+9ff28f4090 via fabric-api
	- fabric-biome-api-v1 9.0.16+83a8659290 via fabric-api
	- fabric-blockrenderlayer-v1 1.1.19+9ff28f4090 via fabric-api
	- fabric-client-tags-api-v1 1.0.0+b35fea8390 via fabric-api
	- fabric-command-api-v1 1.2.10+f71b366f90 via fabric-api
	- fabric-command-api-v2 2.1.6+0c17ea9690 via fabric-api
	- fabric-commands-v0 0.2.27+df3654b390 via fabric-api
	- fabric-containers-v0 0.1.33+df3654b390 via fabric-api
	- fabric-content-registries-v0 3.2.2+83a8659290 via fabric-api
	- fabric-convention-tags-v1 1.1.0+7cd20a1490 via fabric-api
	- fabric-crash-report-info-v1 0.2.4+bd0a0d4a90 via fabric-api
	- fabric-data-generation-api-v1 5.1.7+a680b9b490 via fabric-api
	- fabric-dimensions-v1 2.1.29+a6d2f78590 via fabric-api
	- fabric-entity-events-v1 1.4.17+9ff28f4090 via fabric-api
	- fabric-events-interaction-v0 0.4.27+9ff28f4090 via fabric-api
	- fabric-events-lifecycle-v0 0.2.27+df3654b390 via fabric-api
	- fabric-game-rule-api-v1 1.0.20+1899036190 via fabric-api
	- fabric-item-api-v1 1.5.6+35a03c4390 via fabric-api
	- fabric-item-groups-v0 0.3.27+6bee109e90 via fabric-api
	- fabric-key-binding-api-v1 1.0.19+9ff28f4090 via fabric-api
	- fabric-keybindings-v0 0.2.17+df3654b390 via fabric-api
	- fabric-lifecycle-events-v1 2.1.1+83a8659290 via fabric-api
	- fabric-loot-api-v2 1.1.2+83a8659290 via fabric-api
	- fabric-loot-tables-v1 1.1.5+9e7660c690 via fabric-api
	- fabric-message-api-v1 5.0.2+176380a290 via fabric-api
	- fabric-mining-level-api-v1 2.1.13+33fbc73890 via fabric-api
	- fabric-models-v0 0.3.16+9ff28f4090 via fabric-api
	- fabric-networking-api-v1 1.2.3+5eb68ef290 via fabric-api
	- fabric-networking-v0 0.3.20+df3654b390 via fabric-api
	- fabric-object-builder-api-v1 4.0.10+7675279690 via fabric-api
	- fabric-particles-v1 1.0.9+dc39553a90 via fabric-api
	- fabric-registry-sync-v0 0.9.23+23c4cfef90 via fabric-api
	- fabric-renderer-api-v1 1.0.9+84f79e9d90 via fabric-api
	- fabric-renderer-indigo 0.6.11+5187d39f90 via fabric-api
	- fabric-renderer-registries-v1 3.2.19+df3654b390 via fabric-api
	- fabric-rendering-data-attachment-v1 0.3.13+9ff28f4090 via fabric-api
	- fabric-rendering-fluids-v1 3.0.6+9ff28f4090 via fabric-api
	- fabric-rendering-v0 1.1.21+df3654b390 via fabric-api
	- fabric-rendering-v1 1.10.15+9ff28f4090 via fabric-api
	- fabric-resource-conditions-api-v1 2.0.10+e62f51a390 via fabric-api
	- fabric-resource-loader-v0 0.6.0+6bee109e90 via fabric-api
	- fabric-screen-api-v1 1.0.25+f05effaf90 via fabric-api
	- fabric-screen-handler-api-v1 1.2.12+9ff28f4090 via fabric-api
	- fabric-textures-v0 1.0.19+9ff28f4090 via fabric-api
	- fabric-transfer-api-v1 2.0.10+83a8659290 via fabric-api
	- fabric-transitive-access-wideners-v1 1.2.0+1e232e1390 via fabric-api
	- fabricloader 0.14.9
	- gravity_api 0.7.12
	- imm_ptl_core 2.2.0 via immersive_portals
	- immersive_portals 2.2.0
	- java 17
	- kanos_config 0.3.1+1.14.4-1.19 via pehkui
	- minecraft 1.19.2
	- mini_scaled 2.0.0
	- pehkui 3.4.0+1.14.4-1.19.2
	- q_misc_util 2.2.0 via immersive_portals
	- worldedit 7.2.12+6240-87f4ae1
[03:48:02] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/D:/Games/%d0%be%d0%b1%d0%be%d0%b8/minecraft/libraries/net/fabricmc/sponge-mixin/0.11.4+mixin.0.8.5/sponge-mixin-0.11.4+mixin.0.8.5.jar Service=Knot/Fabric Env=CLIENT
[03:48:02] [main/INFO]: Compatibility level set to JAVA_16
[03:48:02] [main/INFO]: Compatibility level set to JAVA_17
[03:48:04] [main/WARN]: Configuration conflict: there is more than one oshi.properties file on the classpath
[03:48:05] [main/WARN]: Configuration conflict: there is more than one oshi.architecture.properties file on the classpath
[03:48:09] [main/INFO]: Building optimized datafixer
[03:48:17] [Render thread/INFO]: Environment: ElyEnvironment[name=ely,authHost=https://authserver.ely.by/auth,accountsHost=https://account.ely.by/api/mojang,sessionHost=https://account.ely.by/api/minecraft/session,servicesHost=<null>]
[03:48:17] [Render thread/INFO]: CAFixer is not available: java.lang.NoClassDefFoundError: com/turikhay/caf/util/Logger
[03:48:17] [Render thread/INFO]: Setting user: Deluxer
[03:48:18] [Render thread/INFO]: Loading Immersive Portals config
[03:48:18] [Render thread/INFO]: IP Config Applied
[03:48:18] [Render thread/INFO]: Immersive Portals Mod Initializing
[03:48:18] [Render thread/INFO]: Dimensional Threading is not present
[03:48:18] [Render thread/INFO]: Pehkui is present
[03:48:18] [Render thread/INFO]: Gravity API is present
[03:48:18] [Render thread/INFO]: [STDOUT]: MiniScaled Mod Initializing
[03:48:20] [Render thread/INFO]: Got request to register class com.sk89q.worldedit.fabric.FabricPlatform with WorldEdit [com.sk89q.worldedit.extension.platform.PlatformManager@aabb853]
[03:48:20] [Render thread/INFO]: WorldEdit for Fabric (version 7.2.12+6240-87f4ae1) is loaded
[03:48:21] [Render thread/INFO]: [Indigo] Registering Indigo renderer!
[03:48:21] [Render thread/INFO]: [Indigo] Compatibility mode enabled.
[03:48:21] [Render thread/INFO]: Sodium is not present
[03:48:21] [Render thread/INFO]: Iris is not present
[03:48:22] [Render thread/INFO]: Backend library: LWJGL version 3.3.1 SNAPSHOT
[03:48:25] [Render thread/INFO]: Shader Transform Skipping blit_screen
[03:48:25] [Render thread/INFO]: Shader Transform Skipping blit_screen
[03:48:25] [Render thread/INFO]: Shader Transform Skipping blit_screen
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_color
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_color
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_color
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_color_tex
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_color_tex
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_color_tex
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_tex
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_tex
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_tex
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_tex_color
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_tex_color
[03:48:25] [Render thread/INFO]: Shader Transform Skipping position_tex_color
[03:48:25] [Render thread/INFO]: Shader Transform Skipping rendertype_text
[03:48:25] [Render thread/INFO]: Shader Transform Skipping rendertype_text
[03:48:25] [Render thread/INFO]: Shader Transform Skipping rendertype_text
[03:48:25] [Render thread/INFO]: Narrator library for x64 successfully loaded
[03:48:25] [Worker-Main-4/ERROR]: Failed to retrieve profile key pair
java.io.IOException: Could not retrieve profile key pair
	at net.minecraft.class_7434.method_43605(class_7434.java:130) ~[client-intermediary.jar:?]
	at net.minecraft.class_7434.method_44291(class_7434.java:69) ~[client-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) [?:?]
[03:48:25] [Render thread/INFO]: Reloading ResourceManager: Default, Fabric Mods (Cardinal Components API (base), Cardinal Components API (entities), Cloth Config v6, Fabric API, Fabric API Base, Fabric API Lookup API (v1), Fabric Biome API (v1), Fabric BlockRenderLayer Registration (v1), Fabric Client Tags, Fabric Command API (v1), Fabric Command API (v2), Fabric Commands (v0), Fabric Containers (v0), Fabric Content Registries (v0), Fabric Convention Tags, Fabric Crash Report Info (v1), Fabric Data Generation API (v1), Fabric Dimensions API (v1), Fabric Entity Events (v1), Fabric Events Interaction (v0), Fabric Events Lifecycle (v0), Fabric Game Rule API (v1), Fabric Item API (v1), Fabric Item Groups (v0), Fabric Key Binding API (v1), Fabric Key Bindings (v0), Fabric Lifecycle Events (v1), Fabric Loot API (v2), Fabric Loot Tables (v1), Fabric Message API (v1), Fabric Mining Level API (v1), Fabric Models (v0), Fabric Networking API (v1), Fabric Networking (v0), Fabric Object Builder API (v1), Fabric Particles (v1), Fabric Registry Sync (v0), Fabric Renderer API (v1), Fabric Renderer - Indigo, Fabric Renderer Registries (v1), Fabric Rendering Data Attachment (v1), Fabric Rendering Fluids (v1), Fabric Rendering (v0), Fabric Rendering (v1), Fabric Resource Conditions API (v1), Fabric Resource Loader (v0), Fabric Screen API (v1), Fabric Screen Handler API (v1), Fabric Textures (v0), Fabric Transfer API (v1), Fabric Transitive Access Wideners (v1), Fabric Loader, Gravity API, Immersive Portals Core, Immersive Portals, Kanos Config, MiniScaled, Pehkui, QMiscUtil, WorldEdit)
[03:48:25] [Render thread/INFO]: Loaded Shader Code Transformation
[03:48:27] [Realms Notification Availability checker #1/INFO]: Could not authorize you against Realms server: Invalid session id
[03:48:29] [Worker-Main-5/WARN]: Exception loading blockstate definition: 'mini_scaled:blockstates/scale_box_placeholder.json' missing model for variant: 'mini_scaled:scale_box_placeholder#'
[03:48:29] [Worker-Main-5/WARN]: Exception loading blockstate definition: 'mini_scaled:blockstates/barrier.json' missing model for variant: 'mini_scaled:barrier#'
[03:48:31] [Render thread/WARN]: Missing sound for event: minecraft:item.goat_horn.play
[03:48:31] [Render thread/WARN]: Missing sound for event: minecraft:entity.goat.screaming.horn_break
[03:48:31] [Render thread/INFO]: OpenAL initialized on device OpenAL Soft on Динамики (USB Audio Device)
[03:48:31] [Render thread/INFO]: Sound engine started
[03:48:31] [Render thread/INFO]: Created: 1024x1024x4 minecraft:textures/atlas/blocks.png-atlas
[03:48:32] [Render thread/INFO]: Created: 256x128x4 minecraft:textures/atlas/signs.png-atlas
[03:48:32] [Render thread/INFO]: Created: 1024x512x4 minecraft:textures/atlas/banner_patterns.png-atlas
[03:48:32] [Render thread/INFO]: Created: 1024x512x4 minecraft:textures/atlas/shield_patterns.png-atlas
[03:48:32] [Render thread/INFO]: Created: 256x256x4 minecraft:textures/atlas/chest.png-atlas
[03:48:32] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/beds.png-atlas
[03:48:32] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/shulker_boxes.png-atlas
[03:48:34] [Render thread/WARN]: Shader rendertype_entity_translucent_emissive could not find sampler named Sampler2 in the specified shader program.
[03:48:34] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/particles.png-atlas
[03:48:34] [Render thread/INFO]: Created: 256x256x0 minecraft:textures/atlas/paintings.png-atlas
[03:48:34] [Render thread/INFO]: Created: 256x128x0 minecraft:textures/atlas/mob_effects.png-atlas
[03:48:43] [Render thread/INFO]: Registering commands with com.sk89q.worldedit.fabric.FabricPlatform
[03:48:44] [Render thread/INFO]: Loaded 7 recipes
[03:48:45] [Render thread/INFO]: Loaded 1179 advancements
[03:48:58] [Render thread/INFO]: Loaded 7 recipes
[03:48:58] [Render thread/INFO]: Loaded 1179 advancements
[03:48:59] [Worker-Main-6/ERROR]: Failed to retrieve profile key pair
java.io.IOException: Could not retrieve profile key pair
	at net.minecraft.class_7434.method_43605(class_7434.java:130) ~[client-intermediary.jar:?]
	at net.minecraft.class_7434.method_44291(class_7434.java:69) ~[client-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) [?:?]
[03:48:59] [Render thread/INFO]: Applied 0 biome modifications to 0 of 63 new biomes in 3.648 ms
[03:48:59] [Server thread/INFO]: Starting integrated minecraft server version 1.19.2
[03:48:59] [Server thread/INFO]: Generating keypair
[03:49:02] [Server thread/INFO]: Global portal storage initialized minecraft:overworld
[03:49:02] [Server thread/INFO]: Bedrock Replacement minecraft:overworld null
[03:49:02] [Server thread/INFO]: Global portal storage initialized immersive_portals:alternate3
[03:49:02] [Server thread/INFO]: Bedrock Replacement immersive_portals:alternate3 null
[03:49:02] [Server thread/INFO]: Global portal storage initialized minecraft:the_end
[03:49:02] [Server thread/INFO]: Bedrock Replacement minecraft:the_end null
[03:49:02] [Server thread/INFO]: Global portal storage initialized immersive_portals:alternate4
[03:49:02] [Server thread/INFO]: Bedrock Replacement immersive_portals:alternate4 null
[03:49:02] [Server thread/INFO]: Global portal storage initialized immersive_portals:alternate2
[03:49:02] [Server thread/INFO]: Bedrock Replacement immersive_portals:alternate2 null
[03:49:02] [Server thread/INFO]: Global portal storage initialized immersive_portals:alternate1
[03:49:02] [Server thread/INFO]: Bedrock Replacement immersive_portals:alternate1 null
[03:49:02] [Server thread/INFO]: Global portal storage initialized minecraft:the_nether
[03:49:02] [Server thread/INFO]: Bedrock Replacement minecraft:the_nether null
[03:49:02] [Server thread/INFO]: Global portal storage initialized immersive_portals:alternate5
[03:49:02] [Server thread/INFO]: Bedrock Replacement immersive_portals:alternate5 null
[03:49:02] [Server thread/INFO]: Global portal storage initialized mini_scaled:void
[03:49:02] [Server thread/INFO]: Bedrock Replacement mini_scaled:void null
[03:49:02] [Server thread/INFO]: Immersive Portals' Dimension Id Record File Does Not Exist
[03:49:02] [Server thread/INFO]: Immersive Portals' dimension id record is missing
[03:49:02] [Server thread/INFO]: Dimension Id Record is Missing
[03:49:02] [Server thread/INFO]: Start Completing Dimension Id Record
[03:49:02] [Server thread/INFO]: Before:

[03:49:02] [Server thread/INFO]: Server Loaded Dimensions:
immersive_portals:alternate1
immersive_portals:alternate2
immersive_portals:alternate3
immersive_portals:alternate4
immersive_portals:alternate5
minecraft:overworld
minecraft:the_end
minecraft:the_nether
mini_scaled:void

[03:49:02] [Server thread/INFO]: After:
minecraft:overworld -> 0
minecraft:the_nether -> -1
minecraft:the_end -> 1
immersive_portals:alternate1 -> 2
immersive_portals:alternate2 -> 3
immersive_portals:alternate3 -> 4
immersive_portals:alternate4 -> 5
immersive_portals:alternate5 -> 6
mini_scaled:void -> 7
[03:49:02] [Server thread/INFO]: Dimension Id Info Saved to File
[03:49:02] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
[03:49:05] [Render thread/INFO]: Подготовка области возрождения: 0%
[03:49:05] [Render thread/INFO]: Подготовка области возрождения: 0%
[03:49:05] [Render thread/INFO]: Подготовка области возрождения: 0%
[03:49:05] [Render thread/INFO]: Подготовка области возрождения: 0%
[03:49:05] [Render thread/INFO]: Подготовка области возрождения: 0%
[03:49:06] [Render thread/INFO]: Подготовка области возрождения: 0%
[03:49:06] [Render thread/INFO]: Подготовка области возрождения: 0%
[03:49:08] [Render thread/INFO]: Подготовка области возрождения: 0%
[03:49:08] [Render thread/INFO]: Подготовка области возрождения: 0%
[03:49:08] [Render thread/INFO]: Подготовка области возрождения: 0%
[03:49:08] [Render thread/INFO]: Подготовка области возрождения: 0%
[03:49:09] [Render thread/INFO]: Подготовка области возрождения: 8%
[03:49:09] [Render thread/INFO]: Подготовка области возрождения: 8%
[03:49:12] [Render thread/INFO]: Подготовка области возрождения: 15%
[03:49:12] [Render thread/INFO]: Подготовка области возрождения: 15%
[03:49:12] [Render thread/INFO]: Подготовка области возрождения: 15%
[03:49:12] [Render thread/INFO]: Подготовка области возрождения: 15%
[03:49:12] [Render thread/INFO]: Подготовка области возрождения: 15%
[03:49:12] [Render thread/INFO]: Подготовка области возрождения: 15%
[03:49:13] [Render thread/INFO]: Подготовка области возрождения: 62%
[03:49:13] [Render thread/INFO]: Подготовка области возрождения: 62%
[03:49:15] [Render thread/INFO]: Подготовка области возрождения: 69%
[03:49:15] [Render thread/INFO]: Подготовка области возрождения: 69%
[03:49:15] [Render thread/INFO]: Подготовка области возрождения: 69%
[03:49:16] [Render thread/INFO]: Подготовка области возрождения: 86%
[03:49:16] [Render thread/INFO]: Подготовка области возрождения: 86%
[03:49:16] [Server thread/INFO]: Loading custom portal generation
[03:49:16] [Render thread/INFO]: Time elapsed: 12769 ms
[03:49:18] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 2342ms or 46 ticks behind
[03:49:19] [Server thread/INFO]: Server performance level: good
[03:49:19] [Server thread/INFO]: Changing view distance to 9, from 10
[03:49:19] [Server thread/INFO]: Changing simulation distance to 8, from 0
[03:49:22] [Server thread/INFO]: Deluxer[local:E:d97d0dff] logged in with entity id 1 at (7.5, -60.0, -1.5)
[03:49:22] [Server thread/INFO]: Deluxer присоединился к игре
[03:49:24] [Render thread/INFO]: Received Dimension Type Sync
[03:49:24] [Render thread/INFO]: 
immersive_portals:alternate3 -> immersive_portals:surface_type
minecraft:the_end -> minecraft:the_end
minecraft:overworld -> minecraft:overworld
immersive_portals:alternate4 -> immersive_portals:surface_type
immersive_portals:alternate1 -> immersive_portals:surface_type
immersive_portals:alternate2 -> immersive_portals:surface_type
minecraft:the_nether -> minecraft:the_nether
immersive_portals:alternate5 -> immersive_portals:surface_type
mini_scaled:void -> mini_scaled:void_dim_type

[03:49:24] [Render thread/INFO]: Received Dimension Int Id Sync
[03:49:24] [Render thread/INFO]: 
minecraft:overworld -> 0
immersive_portals:alternate1 -> 2
immersive_portals:alternate2 -> 3
minecraft:the_nether -> -1
immersive_portals:alternate3 -> 4
immersive_portals:alternate4 -> 5
minecraft:the_end -> 1
immersive_portals:alternate5 -> 6
mini_scaled:void -> 7
[03:49:24] [Render thread/INFO]: WorldRenderer reloaded minecraft:overworld
[03:49:25] [Render thread/INFO]: Loaded 0 advancements
[03:50:13] [Worker-Bootstrap-3/INFO]: 4283 Datafixer optimizations took 123549 milliseconds
[03:50:40] [Server thread/INFO]: Portal Generation Attempted minecraft:overworld 11 -60 1
[03:50:43] [Render thread/INFO]: Client Teleported Portal{288,west,(minecraft:overworld 11 -59 -1)->(minecraft:overworld 11 -58 2),scale:2.0} 1255
```

**Comment by qouteall** [comment]
I can reproduce this issue now

**Comment by qouteall** [comment]
When debugging this issue suddenly I cannot reproduce this issue anymore. Very weird.

**Comment by qouteall** [comment]
It deadloops inside `Frustum#offsetToFullyIncludeCameraCube`. This method tries to move back the frustum to include the 8x8x8 cube that the camera is in. (this does not interfere with immptl frustum culling code). Normally this method should work and not deadloop.
By testing, I found that without pehkui it won't deadloop. Then I suspect that it's related with pehkui changing the near plane of the projection matrix, then I changed pehkui config to make the near plane not zero, and the issue vanished, however when I change the near plane back to 0 the issue cannot be reproduced.

**Comment by qouteall** [comment]
I simply made the loop to limit to 10 times and it will probably fix the issue.

**Comment by qouteall** [comment]
Does the issue still occur in this version? https://github.com/iPortalTeam/ImmersivePortalsMod/releases/tag/v2.2.1-1.19.2

**Comment by Asaak232** [comment]
> Does the issue still occur in this version? https://github.com/iPortalTeam/ImmersivePortalsMod/releases/tag/v2.2.1-1.19.2



sorry for the delay, and thanks, new version fixed the problem. But there is a bit strange effect, then you go through a portal that changes size, all chanks disappear for a brief moment. I think effect lasted noticeably longer, if you go smaller portal first, and then go back through the portal from which you came out.

**Comment by qouteall** [comment]
@Asaak232 Can you show a video about that?
Does the "chunk disappear" issue occur without Pehkui?
Have you tried deleting pehkui config file and re-launch the game (deleting config file make it to reset config)?

**Comment by Asaak232** [comment]
@qouteall Chunks disappear happens only to portals that uses pehkui, other portals are fine. It works a bit weird, i did some tests, and sometimes all chunks dissapear, but one that beneath me doesnt, as long as i dont touch the mouse, and stay at the same chunk, it will stay like that. Also, sometimes all chunks dissaper, even one that beneath me. But there are still a lot of cases when all chunks disappear only for a moment, or all except the one below me. I tried deleting config, but promblem is still here. 
I recorded video, and here it is https://youtu.be/6nW9bg7Ds5s

**Comment by qouteall** [comment]
I can reproduce this issue again.

**Comment by qouteall** [comment]
The real cause of this issue has been found. The projection matrix's near plane can be set to 0 and its determinant become 0 then frustum related code will malfunction

**Comment by qouteall** [comment]
In Pehkui's `modifyProjectionMatrixDepth`, the player scale is almost 0 in the frame that the player teleports

**Comment by qouteall** [comment]
It seems to be a Pehkui's issue. Right after teleportation `ScaleUtils.getBoundingBoxWidthScale(entity, tickDelta)` returns a very small value. But if tickDelta is 1 it will return a normal value

**Comment by qouteall** [comment]
Pehkui introduces a lot of complexity. The original scale is normal but the scale modifiers make the scale nearly zero. The scale modifiers involve a lot of hard to debug over abstraction code. It contains tons of nesting of calculating scales and contains lamdbas that are hard to debug.

Anyway this is a Pehkui's issue. Calling `ScaleUtils.getEyeHeightScale(entity, 0.5f)` immediately after teleportaion. ImmPtl does not change the scale modification code and this issue does not occur with the old Pehkui

**Comment by Asaak232** [comment]
@qouteall, even so, thanks for figuring that out, but could you please post the issue at the pehkui github? I'm afraid that i will not be able to convey the essence of the problem.

**Comment by Virtuoel** [comment]
This should now be fixed with the Pehkui 3.4.1 release.

**Comment by qouteall** [comment]
The next version of immptl will have "breaks" `"pehkui": "<3.4.1",`
